src/main/java/org/apache/sysds/runtime/matrix/data/LibCommonsMath.java

```diff
@@ -76,11 +79,17 @@ else if (opcode.equals("cholesky"))
 	public static MatrixBlock[] multiReturnOperations(MatrixBlock in, String opcode) {
 		if(opcode.equals("qr"))
 			return computeQR(in);
+		else if (opcode.equals("qr2"))
+			return computeQR2(in);
 		else if (opcode.equals("lu"))
 			return computeLU(in);
 		else if (opcode.equals("eigen"))
 			return computeEigen(in);
-		else if ( opcode.equals("svd"))
+		else if (opcode.equals("eigen_lanczos"))
+			return computeEigenLanczos(in);
+		else if (opcode.equals("eigen_qr"))
+			return computeEigenQR(in);
+		else if (opcode.equals("svd"))
```

Comment:
Hi @Baunsgaard 

how about we document these functions and mention they are experimental (in the doc)?

Reply:
for now no, these operations are only available sideways calling directly into LibCommonsMath